### DISCUSSION TITLE
Can I return a JSX element from createDOM?

### DISCUSSION THREAD

**Discussion by FrancescoSaverioZuppichini** [facebook/lexical#3604] [Q&A] [answered]
Hi guys,

Is it possible to return a JSK element from `createDOM` inside a `Node` instance?

**Comment by trueadm** [comment, marked as answer]
No, you must return an actual DOM element. Lexical's nodes rely on vanilla JS. You can use a DecoratorNode to try and render React content using the `decorate` method, which does allow for a JSX element to be returned. However, DecoratorNodes are intended to be for things like widgets and not for things like normal text – as they are treated like a blackbox and thus do not gain native Lexical selection. 

https://lexical.dev/docs/concepts/nodes#extending-decoratornode
